pkg/api/migrate_machine.go

```diff
@@ -2,41 +2,70 @@ package api
-	TargetUser       string            `json:"target_user,omitempty"`
-	ExcludePaths     []string          `json:"excluded_paths,omitempty"`
-	TcpPorts         map[uint16]uint16 `json:"tcp_ports,omitempty"`
-	ExcludedTcpPorts []uint16          `json:"excluded_tcp_ports,omitempty"`
-	ForceCreate      bool              `json:"force_create,omitempty"`
-	DisableStart     bool              `json:"disable_start,omitempty"`
-	Debug            bool              `json:"debug,omitempty"`
+	StartContainer    bool     `json:"start_container"`
+	ContainerName     string   `json:"container_name"`
+	ForceCreate       bool     `json:"force_create"`
+	SourceHost        string   `json:"source_host"`
+	SourceUser        string   `json:"source_user"`
+	TargetHost        string   `json:"target_host"`
+	TargetUser        string   `json:"target_user"`
+	ExcludePaths      []string `json:"excluded_paths"`
+	UseDefaultPortMap bool     `json:"use_default_port_map"`
 }
 
+// buildActorInput translates the data sent by the client into data that the actor can interpret.
+func buildActorInput(p *MigrateParams) (string, error) {
+	data := map[string]interface{}{
+		"start_container":        ObjValue{p.StartContainer},
+		"container_name":         ObjValue{p.ContainerName},
+		"force_create":           ObjValue{p.ForceCreate},
+		"source_host":            ObjValue{p.SourceHost},
+		"source_user_name":       ObjValue{p.SourceUser},
+		"target_host":            ObjValue{p.TargetHost},
+		"target_user_name":       ObjValue{p.TargetUser},
+		"excluded_paths":         ObjValue{p.ExcludePaths},
+		"use_default_port_map":   ObjValue{p.UseDefaultPortMap},
+		"excluded_tcp_ports":     ExcludedTCPPorts{TCP: make(map[uint16]uint16)},
+		"tcp_ports_user_mapping": TCPPortsUserMapping{[]string{}},
+	}
+
+	j, err := json.Marshal(data)
+	if err != nil {
+		return "", err
+	}
+
+	return string(j), nil
+}
+
+// MigrateMachine handles the /migrate-machine endpoint.
 func MigrateMachine(request *http.Request) (interface{}, error) {
-	var (
-		migrateParams MigrateParams
-		data          interface{}
-	)
+	var params MigrateParams
 
-	if err := json.NewDecoder(request.Body).Decode(&migrateParams); err != nil {
+	if err := json.NewDecoder(request.Body).Decode(&params); err != nil {
 		return nil, err
 	}
 
-	// place for executor
-	// Executor.execute // or something
-	output_mock := []byte("{\"foo\": \"bar\"}")
-	// return result to client
-
-	if err := json.Unmarshal(output_mock, &data); err != nil {
+	// Translate data sent by client into data that actor can read
+	actorInput, err := buildActorInput(&params)
+	if err != nil {
 		return nil, err
 	}
 
-	return data, nil
+	// Call the actor runner passing data to its stdin
+	c := executor.New("migrate-machine", actorInput)
+
+	// TODO: this blocks until the migration is done; so we might execute this in a worker in the future
+	r := c.Execute()
+
+	log.Println(r.Stderr)
+
+	return r.Stdout, nil
```

Comment:
This does not seem right here at all - We have to change this bit - Let's discuss this F2F